components/eam/src/physics/cam/cam_diagnostics.F90

```diff
@@ -1757,6 +1757,8 @@ subroutine diag_phys_writeout(state, psl)
 
 !! Boundary layer atmospheric stability, temperature, water vapor diagnostics
 
+    p_surf_t1 = 0._r8
+    p_surf_t2 = 0._r8
```

Comment:
Nice catch, @amametjanov . In the same spirit, we probably should also initialize p_surf_q1 and p_surf_q2.  vertinterp calculations only loop over ncol. Just like p_surf_t1 and p_surf_t2, p_surf_q1 and p_surf_q2 could also be invoked in full array operation when certain output variable is specified.

Reply:
Thanks, done.